examples/locks/locks.bal

```diff
@@ -0,0 +1,39 @@
+import ballerina/io;
+import ballerina/runtime;
+import ballerina/time;
+
+// Shared variable among multiple workers.
+string sharedString;
+
+function main (string... args) {
+    worker w1 {
```

Comment:
This example looks a bit artificial with the sleep operations and all. And it is not directly clear how the locks are useful at first. Basically, it doesn't show a real-world example. Shall we just do a simple atomic counter implementation scenario, where multiple workers will be incrementing a single integer value, or some type of a similar scenario.